src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -512,16 +513,32 @@ public FormValidation doTestConnection(@QueryParameter String name, @QueryParame
                         Util.fixEmpty(serverCertificate), Util.fixEmpty(credentialsId), skipTlsVerify,
                         connectionTimeout, readTimeout).createClient();
 
+                // test listing pods
                 client.pods().list();
-                return FormValidation.ok("Connection successful");
+
+                // test creating pods
+                Pod pod = client.pods().create(new PodBuilder() //
+                        .withNewMetadata().withGenerateName("kubernetes-plugin-").endMetadata() //
+                        .withNewSpec() //
+                        .withContainers( //
+                                new ContainerBuilder().withName("alpine").withImage("alpine").withCommand("cat")
```

Comment:
This will fail if no "alpine" image is available.

Jenkins might be running in a closed build environment which does not have access to dockerhub, but only to a private registry with custom images. May happen in an enterprise setting..

Reply:
it should should the correct error in that case